logstash-core/src/main/java/org/logstash/ackedqueue/FileSettings.java

```diff
@@ -0,0 +1,79 @@
+package org.logstash.ackedqueue;
+
+import org.logstash.common.io.CheckpointIOFactory;
+import org.logstash.common.io.PageIOFactory;
+
+public class FileSettings implements Settings {
```

Comment:
yes, I think I agree here - having a Settings interface and specific Settings implementations seems not required here since both FileSettings and MemorySettings are actually identical objects that differ only in the constructor. 
+1
@guyboertje authored these bits, WDYT?